### PR TITLE
provide Changelog and What's New information

### PR DESCRIPTION
**Pull request by blizzz**
Solves #115 and needed for https://github.com/nextcloud/server/issues/9163 

@MorrisJobke @nickvergessen are you ok with that approach? If so, don't merge yet, the example in the productive whatsnew.json file should be replaced with something reasonable first :)

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Also the whatsnew.json is just a plain JS file that is requested or how does it work? Is this integrated in the XML? because then it could be that the request get's quite big.

File: config/whatsnew.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "13.0.1" : {
+    "en" : [
+      "Pollute the files failing to update l n files for pdf",
+      "Emaillogin server show open graph preview in gallery removed old code",
+      "Server sharee email search server log exceptions that s settings server"
```
Comment (MorrisJobke) [inline config/whatsnew.json]: Why not ship markdown directly? Then we could also format it a bit better.

Comment (MorrisJobke) [reply config/whatsnew.json]: But has the downside of a potential code injection.

**Comment by blizzz** [comment]
>Also the whatsnew.json is just a plain JS file that is requested or how does it work? Is this integrated in the XML? because then it could be that the request get's quite big.

Only the text relevant to the version is returned with the response.

File: config/whatsnew.json (hunk 1 above)
Comment (blizzz) [reply config/whatsnew.json]: It's safer,  and simpler. Imho, thee short lines can live well without formatting. @jancborchardt ?

**Comment by rullzer** [comment]
Needs rebase

File: config/whatsnew.json (hunk 1 above)
Comment (jancborchardt) [reply config/whatsnew.json]: We want to display it really simply, boiled down to the essence. Like in the updates screen of Android Play Store or iOS Appstore. There’s no formatting.

**Comment by MorrisJobke** [comment]
The response then looks like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<nextcloud>
 <version>13.0.2.1</version>
 <versionstring>Nextcloud 13.0.2</versionstring>
 <url>https://download.nextcloud.com/server/releases/nextcloud-13.0.2.zip</url>
 <web>https://docs.nextcloud.com/server/13/admin_manual/maintenance/upgrade.html</web>
 <changelog>https://nextcloud.com/changelog/#13-0-2</changelog>
 <autoupdater>1</autoupdater>
 <eol>0</eol>
 <signature>LM7g6SaWxzArRLd++9sdk80UZw1mIzAaqEnH75YO3EBK7qF7xxVdvb7tnu9cipsr
Y5e/MoISsNOgc3FZoRygGrkKl4GVf4SZSOSusmoDxwHNT0tBSK8u0QMRXrEXLuyE
pxyvA3hHMP/gGhLn+T0/CrvI/QBy3QO1N74OUWc+nc+NnKnWSwwVP4/OITRkAtYj
zsIH95cgI4e3Eez6c63YsITKLWVwoE4Qu0ubU8HGfhHBYxqFoGbNfSWvB78gV/gf
S3QFUmoMjKoOvo1Q4s7UQv0xE5TUDC4XFIW6q1ly1Z/XRPZaywstv3br8J52L6+J
G2ZuS4ms5s9o5i2OrogYRQ==</signature>
 <whatsNew_admin>
  <item>Pollute the files failing to update l n files for pdf</item>
  <item>Emaillogin server show open graph preview in gallery removed old code</item>
  <item>Server sharee email search server log exceptions that s settings server</item>
 </whatsNew_admin>
 <whatsNew_regular>
  <item>A beautifully revamped web UI</item>
  <item>Sparkling buttons with pew-pew-sound effects when you look at them</item>
  <item>Everything is shared automatically with your mother-in-law</item>
 </whatsNew_regular>
</nextcloud>
```

**Comment by MorrisJobke** [comment]
As discussed:

I guess that adding this to the default response causes a huge amount of traffic. Thus I would love to have something like:

`https://updates.nextcloud.com/changelog/server/13.0.3` with an accept header  of the user language to send over then only the changelog with only that matches those criteria. Also then a transifex integration is not too hard as well.

**Comment by blizzz** [comment]
As discussed with @MorrisJobke today:

* serve the infos from a new server instread (easier to seperate)
* instead of supplying the what's new info, specify an URL to a separate endpoint
* then, deliver all translations, so the Nextcloud server can present it accordingly
* provide an etag over the content for proper caching (and keep traffic low)

**Comment by jancborchardt** [comment]
Also:
- Of course the whats new infos are optional
- And admins should see the same as users. But there can be optional additional infos for admins on top.

**Comment by blizzz** [comment]
@MorrisJobke @rullzer I updated the PR to only provide a changes field for versions >= 14.0.0.

with test data:

screenshot_20180615_180219

whether the url exists is not being checked here. Client needs to deal with that also for various other reasons.

**Comment by blizzz** [comment]
and now even the tests pass :)

**Pull request merged**